test/elasticsearch/drain/test_autoscaling.rb

```diff
@@ -3,15 +3,78 @@
 
 class TestAutoScaling < Minitest::Test
   def setup
-    VCR.insert_cassette 'autoscaling', record: :new_episodes
-    @asg = ::Elasticsearch::Drain::AutoScaling.new(
-      'esuilogs-razor-d0prod-r01-v000',
-      'us-east-1'
-    )
+    Aws.config[:stub_responses] = true
+    @asg = ::Elasticsearch::Drain::AutoScaling.new('my-asg', 'us-east-1')
+    stub_describe_auto_scaling_groups(@asg.asg_client, ['i-abcd1234', 'i-1234abcd'], 'my-asg')
+    stub_describe_auto_scaling_instances(@asg.asg_client, ['i-abcd1234', 'i-1234abcd'], 'my-asg')
+    stub_ec2_describe_instances(@asg.ec2_client, 'i-abcd1245' => '192.168.0.3', 'i-1234abcd' => '192.168.0.4')
   end
 
-  def teardown
-    VCR.eject_cassette
+  def stub_describe_auto_scaling_groups(asg_client, instances, asg_name)
+    raise ArgumentError, 'instances must be an array' unless instances.respond_to?(:each)
+
+    launch_config_name = "#{asg_name}-#{SecureRandom.hex[0..10].upcase}"
+    new_instances = []
+    instances.each do |instance|
+      instance_hash = instance_hash(instance, availability_zones.sample)
+      instance_hash[:launch_configuration_name] = launch_config_name
+      new_instances << instance_hash
+    end
+
+    asg_client.stub_responses(:describe_auto_scaling_groups,
+                              auto_scaling_groups: [{
+                                auto_scaling_group_name: asg_name,
+                                min_size: instances.length,
+                                max_size: instances.length,
+                                desired_capacity: instances.length,
+                                default_cooldown: 300,
+                                availability_zones: ['us-east-1b', 'us-east-1a', 'us-east-1e', 'us-east-1d'],
+                                health_check_type: 'EC2',
+                                created_time: Time.now,
+                                instances: new_instances
+                              }])
+  end
+
+  def stub_describe_auto_scaling_instances(asg_client, instances, asg_name)
```

Comment:
Should we call these methods `stub_asg_*`? They'd then be consistent with the `ec2*` helpers.